### DISCUSSION TITLE
[RSuite v5] 可以考虑调整一下 `Form` `onSubmit` 属性回参吗

### DISCUSSION THREAD

**Discussion by vingeraycn** [rsuite/rsuite#1596] [Q&A]
在 `onSubmit` 回调中，比起 `checkStatus` 其实更多时候更关心 `formValue`，可以考虑将 `formValue` 作为 `onSubmit` 的第一个参数吗？

(或者说为 `Form` 增加一种非受控的用法，`onSubmit?.(formValue ?? _formValue, event: React.FormEvent<HTMLFormElement>)`。表单校验不通过，则不触发 `onSubmit`

<img width="741" alt="图片" src="https://user-images.githubusercontent.com/19839331/112928258-dc063800-9148-11eb-88f0-e5eab96a77f2.png">

[Form](https://github.com/rsuite/rsuite/blob/next/src/Form/Form.tsx#L65)

**Comment by simonguo** [comment]
`formValue` 是可以通过 `onChange`，实时获取最新的值。 
`onSubmit` 是对 form 触发了原生 submit 事件的回调，所以不管表单是否检验通过，最好是都触发，在回调中处理业务逻辑。

你想在 onSubmit 中拿 formValue 的主要场景是什么？

**Reply by vingeraycn** [reply]
> 你想在 onSubmit 中拿 formValue 的主要场景是什么？

对于简单的表单来说，单独维护一个 formValue State 感觉有些麻烦，如果能在 onSubmit 里拿到 formValue 会比较方便
